### PR TITLE
chore(userspace/libsinsp): remove mesos fields from `--list`

### PR DESCRIPTION
**Pull request by therealbobo**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

/kind cleanup

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

/area libsinsp

**Does this PR require a change in the driver versions?**

**What this PR does / why we need it**:

The mesos support has been deprecated: the mesos fields should no longer appear using `--list`.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: userspace/libsinsp/gen_filter.cpp [hunk 1]
```diff
@@ -273,6 +273,12 @@ bool gen_event_filter_factory::filter_field_info::is_skippable()
 	return (tags.find("EPF_TABLE_ONLY") != tags.end());
 }
 
+bool gen_event_filter_factory::filter_field_info::is_deprecated()
+{
+	// Skip fields with the EPF_TABLE_ONLY flag.
```
[the remark this answers is not in the document]
Comment (therealbobo) [reply userspace/libsinsp/gen_filter.cpp:278]: Ops! 😅

File: userspace/libsinsp/gen_filter.cpp [hunk 2]
```diff
@@ -427,6 +439,11 @@ std::string gen_event_filter_factory::filter_fieldclass_info::as_string(bool ver
 		os << std::endl;
 	}
 
+	if(deprecated_count == fields.size())
```
Comment (FedeDP) [inline userspace/libsinsp/gen_filter.cpp:442]: Why do we return an empty string here? 
We don't do that for `skippable` ones :/

Also, do you think we could merge the `is_deprecated` into `is_skippable` (ie by adding an additional check on `EPF_DEPRECATED`)?
I am worried that we might use the same `is_skippable` method in other part of the code, and managing the new flag in the old method would allow us to skip deprecated fields everywhere for free.

Comment (therealbobo) [reply userspace/libsinsp/gen_filter.cpp:442]: The idea behind this is that if all the fields are deprecated, the field class is deprecated as well: in that case we don't want to print out the field class at all.

I think could be a good idea but I have a concern: if all fields are EPF_TABLE_ONLY, should we omit the field class from the `--list`?

Comment (FedeDP) [reply userspace/libsinsp/gen_filter.cpp:442]: > The idea behind this is that if all the fields are deprecated, the field class is deprecated as well: in that case we don't want to print out the field class at all.

Makes sense!

> I think could be a good idea but I have a concern: if all fields are EPF_TABLE_ONLY, should we omit the field class from the --list?

I am not sure :/ 
I checked and `is_skippable` is used only twice; i think we can manage to keep the new `is_deprecated` method so that it is much cleaner too.

Also, should we update `as_markdown` method too?

Comment (therealbobo) [reply userspace/libsinsp/gen_filter.cpp:442]: Sure! I totally forgot about that! 😄

**Comment by LucaGuerra** [comment]
I agree with the idea. Basically some features are being deprecated and, for example, in Falco we don't want to document them in the help.

One question is, what if a client does use the deprecated fields and wants to document them? Maybe we could add an additional parameter (optional, defaulting to the behavior above) to control this behavior.

Re. markdown, to make sure it's right this is where it ultimately ends up:
https://github.com/falcosecurity/falco-website/blob/master/content/en/docs/reference/rules/supported-fields/index.md?plain=1

You can simply check that the generated md makes sense when inserted in this document.

**Comment by LucaGuerra** [comment]
If the field is "hard deprecated" it shouldn't exist in the first place :)

**Comment by therealbobo** [comment]
You're right: it's a bit confusing. Let's stick with EPF_DEPRECATED and another enum (e.g. EPF_DOCUMENTED).

File: userspace/libsinsp/gen_filter.cpp [hunk 3]
```diff
@@ -302,9 +308,10 @@ void gen_event_filter_factory::filter_fieldclass_info::wrapstring(const std::str
 	}
 }
 
-std::string gen_event_filter_factory::filter_fieldclass_info::as_markdown(const std::set<std::string>& event_sources)
+std::string gen_event_filter_factory::filter_fieldclass_info::as_markdown(const std::set<std::string>& event_sources, bool include_deprecated)
```
Comment (LucaGuerra) [inline userspace/libsinsp/gen_filter.cpp:311]: I think the usage of `incude_deprecated` is missing in this function :thinking:

Comment (therealbobo) [reply userspace/libsinsp/gen_filter.cpp:311]: You are right! 😅

**Comment by LucaGuerra** [comment]
I have restarted the tests, if they pass I think we can merge this!

File: userspace/libsinsp/event.h [hunk 4]
```diff
@@ -63,7 +64,7 @@ typedef enum filtercheck_field_flags
 typedef struct filtercheck_field_info
 {
 	ppm_param_type m_type; ///< Field type.
-	filtercheck_field_flags m_flags;  ///< Field flags.
+	uint32_t m_flags;  ///< Field flags.
```
Comment (FedeDP) [inline userspace/libsinsp/event.h:67]: Why this change?

**Review by leogr — APPROVED** [review]
